### ISSUE TITLE
Pilot functionality to enable the tracking for timeout budgets

### ISSUE THREAD

**Issue by vladg9** [istio/istio#22160]
**Describe the feature request**

TL;DR Add environment variable in Pilot for turning on the tracking of timeout budgets using Envoy's metrics by setting (this)[https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cluster.proto#envoy-api-field-cluster-track-timeout-budgets] to `true`.

Envoy has some nice [metrics](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cluster.proto#envoy-api-field-cluster-track-timeout-budgets) for following the percentage of global timeout and per try timeout budget usage. This statistics would be useful for people who utilise Virtual Services. However, this is (disabled by default)[https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cluster.proto#envoy-api-field-cluster-track-timeout-budgets] and Pilot currently is not able to configure this. It would be nice to be able to turn this on in a similar way the `use_remote_addr` is configured using environment variables in Pilot.

**Describe alternatives you've considered**

Doing this on application level but it seems very inconvenient because we would like to use Virtual Services to abstract things like timeouts away from application level code but to create the needed metrics then the timeouts will have to be visible to the application.

**Additional context**

I am okay with taking on this feature if it's accepted.

**Comment by howardjohn** [comment]
We support modifying arbitrary Envoy configs: https://istio.io/docs/reference/config/networking/envoy-filter/

**Comment by vladg9** [comment]
Apologies then, closing the issue. Thanks!
